pandas/tseries/index.py

```diff
@@ -1581,10 +1590,14 @@ def insert(self, loc, item):
         """
 
         freq = None
+
+        if isinstance(item, np.datetime64):
+            # np.datetime64 doesn't have tz, once convert to Timestamp
+            item = Timestamp(item)
```

Comment:
actually if you always wrap other then you can simply return ``Timestamp(other).tz`` I think

(though may be overlooking something)